web/modules/custom/self_deposit/src/Form/SelfDepositCreateForm.php

```diff
@@ -0,0 +1,231 @@
+    $values = $webform_submission->getData();
+    $files = $values['file'];
+    $file_repository = \Drupal::service('file.repository');
+    $new_dest = "fedora://c160/";
+
+    if (count($files) > 1) {
+      $model = 'Complex Object';
+      $child_files = [];
+      foreach ($files as $file_id) {
+        $file = $this->entityTypeManager->getStorage('file')->load(intval($file_id));
+        $mime = $file->getMimeType();
+        $filename = $file->getFilename();
+        $file = $file_repository->copy($file, $new_dest . $filename);
+        $file_id = $file->id();
+        list($fmodel, $fmedia_type, $ffield_name) = $this->depositUtils->getModel($mime, $filename);
+        $child_files[$file_id] = [
+          'model' => $fmodel,
+          'media_type' => $fmedia_type,
+          'field_name' => $ffield_name,
+          'file_name' => $filename,
+        ];
+      }
+    }
+    else {
+      $file = $this->entityTypeManager->getStorage('file')->load(intval($files[0]));
+      $mime = $file->getMimeType();
+      $filename = $file->getFilename();
+      $files[0] = $file_repository->copy($file, $new_dest . $filename)->id();
+      list($model, $media_type, $field_name) = $this->depositUtils->getModel($mime, $filename);
+    }
+
+    $term = $model;
+    $taxo_manager = $this->entityTypeManager->getStorage('taxonomy_term');
+    $taxo_terms = $taxo_manager->loadByProperties(['name' => $term]);
+    $taxo_term = reset($taxo_terms);
+
+    $copyright_term_arr = $taxo_manager->loadByProperties(['name' => 'In Copyright']);
+    $copyright_term = reset($copyright_term_arr);
+
+    $perm_term_arr =
+    $taxo_manager->loadByProperties(['name' => $values['file_permissions_select']]);
+    $perm_term = reset($perm_term_arr);
+
+    $config = $this->configFactory->get('self_deposit.selfdepositsettings');
+    if ($config->get('collection_for_deposits')) {
+      $member_of = $config->get('collection_for_deposits');
+    }
+
+    if ($model == 'Complex Object') {
+      $node = $this->createNode($webform_submission, $values, $values['item_title'], $taxo_term, $copyright_term, $perm_term, $member_of);
+      foreach ($child_files as $cfkey => $cfvalues) {
+        $fmember_of = $node->id();
+        $ftaxo_terms = $taxo_manager->loadByProperties(['name' => $cfvalues['model']]);
+        $ftaxo_term = reset($ftaxo_terms);
+        $child_node = $this->createNode($webform_submission, $values, $cfvalues['file_name'], $ftaxo_term, $copyright_term, $perm_term, $fmember_of);
+        $this->depositUtils->createMedia($cfvalues['media_type'], $cfvalues['field_name'], $cfkey, $child_node->id());
+      }
+    }
+    else {
+      $node = $this->createNode($webform_submission, $values, $values['item_title'], $taxo_term, $copyright_term, $perm_term, $member_of);
+      $this->depositUtils->createMedia($media_type, $field_name, $files[0], $node->id());
+    }
+
+    $webform_submission->setElementData('item_node', $node->id());
```

Comment:
This doesn't appear to set a value. Do you need to do something like `$webform_submission->save();` for it to persist?

Reply:
This whole function was copy-pasted from the existing webform handler. I'm not sure why it's there, that whole section could probably use a cleanup, but it also seems to work.